model/cachedfeed.py

```diff
@@ -78,14 +80,9 @@ def fetch(cls, _db, lane, type, facets, pagination, annotator,
         from ..opds import AcquisitionFeed
         from ..lane import Lane, WorkList
         if max_age is None:
-            if type == cls.GROUPS_TYPE:
-                max_age = AcquisitionFeed.grouped_max_age(_db)
-            elif type == cls.PAGE_TYPE:
-                max_age = AcquisitionFeed.nongrouped_max_age(_db)
-            elif hasattr(lane, 'MAX_CACHE_AGE'):
-                max_age = lane.MAX_CACHE_AGE
-            else:
-                max_age = 0
+            # Figure out the default max age for a feed of this type
+            # for this lane.
+            max_age = cls.calculate_max_age(_db, lane, type)
```

Comment:
I was getting test failures in circulation because of this code, and I think this code is wrong in a world where there are lots of custom WorkLists which all define MAX_CACHE_AGE. But it didn't have good test coverage. I moved it into a separate method, implemented the behavior I think is right (although a lot of this behavior will probably never happen in real life), and tested it.